packages/katello/rubygem-pulp_container_client/rubygem-pulp_container_client.spec

```diff
@@ -1,26 +1,30 @@
-# Generated from pulp_docker_client-4.0.0b6.dev01562331743.gem by gem2rpm -*- rpm-spec -*-
 # template: scl
 %{?scl:%scl_package rubygem-%{gem_name}}
 %{!?scl:%global pkg_name %{name}}
 
-%global gem_name pulp_docker_client
+%global gem_name pulp_container_client
+
+%global release 1
+%global prerelease rc1
+%global prereleaserpm %{?prerelease:.}%{?prerelease}
```

Comment:
note in other places i see we use prereleasesource and prerelease, but the macro that defines gem_instdir uses prerelease, so that model won't work

Reply:
We use this in the katello gem:
https://github.com/theforeman/foreman-packaging/blob/6f54cca5530a9ad10bc545aa86ff51f05aadcada/packages/katello/rubygem-katello/rubygem-katello.spec#L8-L9

Reply:
@evgeni what do you think about including this by default in our gem2rpm templates?

Reply:
that didn't work here because the source was '3.0.0rc1' and NOT '3.0.0.rc1'

Reply:
Do we have control over that? Gem has built-in support for prerelease when it finds a dash in the version (https://github.com/Katello/katello/blob/master/lib/katello/version.rb#L2) which it translates to a `.` and then some builtin mechanisms handle this properly in the RPMs which is how Katello and hammer gems work.

Reply:
Also, I am good with solving this for now this way. But if we can improve this for the future for consistency with our other gems that would be nice.

Reply:
Eric said everything I wanted to say, but I'll repeat, just for the sake of my own understanding:
* ruby gems (usually) use `.rc1` or similar suffixes to denote a pre-release version, even tho `rc1` (without the dot) would be sufficient (see https://guides.rubygems.org/patterns/#prerelease-gems, any character triggers "prerelease" mode)
* the generated gems use a more pythonic-aproach (https://www.python.org/dev/peps/pep-0440/#pre-releases) without the dot
* the rubygems rpm macros (which we don't control) allow to use the version without the dot too.
* it would be awesome if we could use the same template for every gem, which would require the pulp gems to switch to include the dot
* the current state is perfectly fine to go with for now

Reply:
@evgeni @ekohl sooooo ACK?   :)